File: leetcode/easy/0628. Maximum Product of Three Numbers/main.ts
function maximumProduct(nums: number[]): number {
  nums.sort((a, b) => a - b);

  const n = nums.length;

  const first = nums[0] * nums[1] * nums[n-1];
  const second = nums[n-1] * nums[n-2] * nums[n-3];

  return Math.max(first, second);
};

File: leetcode/easy/0628. Maximum Product of Three Numbers/main.rs
impl Solution {
    pub fn maximum_product(mut nums: Vec<i32>) -> i32 {
        nums.sort();

        let n = nums.len();

        let first = nums[0] * nums[1] * nums[n - 1];
        let second = nums[n - 1] * nums[n - 2] * nums[n - 3];

        if first > second {
            first
        } else {
            second
        }
    }
}
